modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/index.js

```diff
@@ -156,6 +157,98 @@ export async function loadData(
 		);
 	}
 
+	if (Liferay.FeatureFlags['LPD-25230'] && queryString) {
+		const queryStringArray = queryString.split('&');
+
+		queryStringArray.forEach((parameter) => {
+			const [key, value] = parameter.split('=');
+
+			if (key === 'filter' && url.searchParams.get('filter')) {
+
+				// Add new filters to the end of the existing filter.
+
+				const existingFilters = url.searchParams.get('filter');
+
+				url.searchParams.delete('filter');
+
+				url.searchParams.append(
+					'filter',
+					existingFilters + ' and ' + value
+				);
+			}
+			else if (key === 'sort' && url.searchParams.get('sort')) {
+
+				// Add new sort values to the end of the existing sort.
+				// Don't add new sort values for any already existing fields.
+
+				const newSortParams = [];
+
+				const existingSortArray = url.searchParams
+					.get('sort')
+					.split(',');
+
+				const existingSortParamFields = existingSortArray.map(
+					(sort) => sort.split(':')[0]
+				);
+
+				const queryStringSortValueArray = value.split(',');
+
+				queryStringSortValueArray.forEach(
+					(queryStringSortValueItem) => {
+						if (
+							!existingSortParamFields.includes(
+								queryStringSortValueItem.split(':')[0]
+							)
+						) {
+							newSortParams.push(queryStringSortValueItem);
+						}
+					}
+				);
```

Comment:
Good question Marko. To me, this needs a bit of consideration. Because the original intent is to have "Additional parameters", it's not clear to me that those mean "merge" vs "override".

At the very least, I'd consider "filter" as a special case where "merge" is desirable. For the rest, we can just have an override. Let's talk about this early morning next week so that we can finish this.

Reply:
See https://github.com/liferay-frontend/liferay-portal/pull/4405#discussion_r1761269382 I wanted to highlight most relevant code.